### PR TITLE
Make message enumerator timeout configurable

### PR DESCRIPTION
**Pull request by andreasohlund**

### CODE REVIEW

**Comment by andreasohlund** [comment]
@ramonsmits I guess you still want the log changes included as well?

**Comment by andreasohlund** [comment]
Note that this PR changes the default to `500ms`, are we happy with that?

**Comment by danielmarbach** [comment]
@andreasohlund I don't know. It is already 50 times better than before so I guess why not. I'm wondering more if we should override this value for the acceptance tests and set it there to 10 milliseconds to speed up the test execution again.

**Comment by andreasohlund** [comment]
Good point, thoughts on that @ramonsmits ?

**Comment by andreasohlund** [comment]
@danielmarbach I've made the tests use `10ms` as the timeout

File: src/NServiceBus.Transport.Msmq.AcceptanceTests/ConfigureEndpointMsmqTransport.cs [hunk 1]
```diff
@@ -13,10 +13,11 @@ public class ConfigureEndpointMsmqTransport : IConfigureEndpointTestExecution
     public Task Configure(string endpointName, EndpointConfiguration configuration, RunSettings settings, PublisherMetadata publisherMetadata)
     {
         queueBindings = configuration.GetSettings().Get<QueueBindings>();
-     
+
         var transportConfig = configuration.UseTransport<MsmqTransport>();
         transportConfig.DisableConnectionCachingForSends();
-        
+        configuration.GetSettings().Set("MessageEnumeratorTimeout", TimeSpan.FromMilliseconds(10));
```
Comment (danielmarbach) [inline src/NServiceBus.Transport.Msmq.AcceptanceTests/ConfigureEndpointMsmqTransport.cs]: Do we need the same for the transport tests?

Comment (andreasohlund) [reply src/NServiceBus.Transport.Msmq.AcceptanceTests/ConfigureEndpointMsmqTransport.cs]: yea, fixed

**Comment by andreasohlund** [comment]
image

We're now back on par with develop when it comes to how long the ATT's take 👍

File: src/NServiceBus.Transport.Msmq/MsmqSettings.cs [hunk 2]
```diff
@@ -50,6 +51,11 @@ public MsmqSettings(ReadOnlySettings settings)
                 TimeToReachQueue = timeToReachQueue;
             }
 
+            if (settings.TryGet<TimeSpan>("MessageEnumeratorTimeout", out var messageEnumeratorTimeout))
```
Comment (danielmarbach) [inline src/NServiceBus.Transport.Msmq/MsmqSettings.cs]: One last thought. I think we should do something like NserviceBus.Transport.Msmq.Key to make it clear it belongs to msmq

**Comment by andreasohlund** [comment]
:+1: fixed now

**Comment by danielmarbach** [comment]
You forgot to change the other keys in ATTs and transport tests

File: src/NServiceBus.Transport.Msmq.AcceptanceTests/ConfigureEndpointMsmqTransport.cs (hunk 1 above)
Comment (danielmarbach) [reply src/NServiceBus.Transport.Msmq.AcceptanceTests/ConfigureEndpointMsmqTransport.cs]: Here

**Comment by andreasohlund** [comment]
LOL, forgot to hit save in VS :facepalm:

**Pull request merged**